### PR TITLE
Bulk Send

### PR DESCRIPTION
**Pull request by dbuscaglia**
### Overview

Bulk capabilities provided for the Notification system.
### Technical changes
- builder interface for bulk sends
- bulk notification model creation happens asynchronously
- receiver no longer is required to persist the notification, making system more robust
### GIF

image
### Reviewers
- [x] @jcomo  
- [x] @hashamali

### CODE REVIEW

File: app/src/main/java/it/quip/android/model/Notification.java [hunk 1]
```diff
@@ -248,6 +289,17 @@ public JSONObject toJson() {
...
+            deliver();
+        } else {
+            // bulk
+            NotificationBatchJobData jobData = new NotificationBatchJobData(this.mRecepients, this.mSenderUid, this);
+            AsyncTask task = new NotificationBatchJob().execute(jobData);
+        }
+
+    }
+
+    private void deliver() {
```
Comment (hashamali) [inline app/src/main/java/it/quip/android/model/Notification.java:302]: `sendOne`?

File: app/src/main/java/it/quip/android/model/Notification.java [hunk 2]
```diff
@@ -248,6 +289,17 @@ public JSONObject toJson() {
     }
 
     public void send() {
+        if (this.mRecepients == null) {
+            deliver();
+        } else {
+            // bulk
```
Comment (hashamali) [inline app/src/main/java/it/quip/android/model/Notification.java]: super detailed comment here, i see

Comment (dbuscaglia) [inline app/src/main/java/it/quip/android/model/Notification.java]: lol #thangs

File: app/src/main/java/it/quip/android/model/Notification.java (hunk 1 above)
Comment (dbuscaglia) [inline app/src/main/java/it/quip/android/model/Notification.java:302]: Hold on this was what I was trying to warn you about. I am pushing a change.

Comment (hashamali) [inline app/src/main/java/it/quip/android/model/Notification.java:302]: thangs

**Pull request merged**